### ISSUE TITLE
Scaling in `_attention`

### ISSUE THREAD

**Issue by nboyd** [bytedance/Protenix#159]
https://github.com/bytedance/Protenix/blob/3e25dec8ea57fd2e287f970589e465b229d0db68/protenix/model/modules/primitives.py#L260

It seems the two different attention implementations here are semantically different: F.scaled_dot_product_attention will scale the logits, while the second implementation will not. 
Is this intentional?
